src/amr/data/field/coarsening/default_field_coarsener.hpp

```diff
@@ -1,6 +1,9 @@
 #ifndef PHARE_DEFAULT_FIELD_COARSENER_HPP
 #define PHARE_DEFAULT_FIELD_COARSENER_HPP
 
+
+#include "core/def/mpi.hpp"
```

Comment:
it's either this or we modify the CXXFLAGS, which I think is not my preferred approach

basically anywhere samrai is imported we need to configure somethings before it, includes it's own config header so we don't get things defined there from 
![image](https://github.com/PHAREHUB/PHARE/assets/30491/39b296f6-3db9-448e-9b39-486697764ced)

alternatively it could be a phare samrai def hjeader, which internally calls the mpi def header

Reply:
This can be seen compiling/linking with caliper [here](https://hephaistos.lpp.polytechnique.fr/teamcity/buildConfiguration/Phare_Phare_Users_Phil_Build/79071?buildTab=log&focusLine=0&logView=flowAware)